### PR TITLE
Increase flow coverage `lib/cli.js`, `lib/lintSource.js`, and `lib/needlessDisables.js`

### PR DESCRIPTION
**Pull request by ivanzusko**
> Fresh PR instead of [https://github.com/stylelint/stylelint/pull/2661](https://github.com/stylelint/stylelint/pull/2661)

- [x] increase flow coverage in lib/lintSource.js from 74.59 to 79.68%
- [x] increase flow coverage in lib/needlessDisables.js from 74.47 up to 97.28%
- [x] increase flow coverage in lib/cli.js from 55.06 up to 76.31%

### CODE REVIEW

**Comment by ivanzusko** [comment]
@ntwb I'm sorry... do not understand why the build is falling down 😞

**Comment by ntwb** [comment]
There's some duplicated code [here](https://github.com/ivanzusko/stylelint/blob/1c6f19892a44cdfb729e31382f305657e9dab78e/lib/cli.js#L350-L366), was that intentional?

**Comment by ivanzusko** [comment]
@ntwb my bad... don't understand how that could have happened..

**Comment by ivanzusko** [comment]
@ntwb still falling down... Maybe because of `[eslint] Extra semicolon. (semi)` errors ?

**Comment by jeddy3** [comment]
The errors [seem flow related](https://travis-ci.org/stylelint/stylelint/jobs/272391183). @ivanzusko Are you able to rebase of `master` to ensure everything is up to date, and then investigate those errors?

**Comment by CAYdenberg** [comment]
In `lib/needlessDisables.js`, Flow is erroring on line 86 when a "range" is pushed into the `unused.range`. This is because in `decls/stylelint.js`

```js
export type stylelint$needlessDisablesReport = Array<{
  source: string,
  ranges: Array<{
    source: string,
    ranges: Array<{
      start: number,
      end?: number,
    }>,
  }>,
}>
```

So I think Flow has actually caught a bug, or at least something bizarre. The faulty line is pushing into ranges, but it's the ranges within the ranges that have the right data type.

Anyway, I think someone who has better familiarity with the intention behind this code should chime in on that type declaration - I would at least recommend renaming something.

File: lib/lintSource.js [hunk 1]
```diff
@@ -8,6 +8,124 @@ const getOsEol = require("./utils/getOsEol");
...
+      input: {
+        file: ?string,
+      },
+    },
+  },
+  messages: Array<any>,
+  stylelint: Object,
+}*/
+
+/*:: type postcssResultT = {
```
Comment (jeddy3) [inline lib/lintSource.js]: Dupe?

File: lib/lintSource.js [hunk 2]
```diff
@@ -8,6 +8,124 @@ const getOsEol = require("./utils/getOsEol");
...
+  stylelint: {
+    customMessages: Object,
+    ignored?: boolean,
+    ignoreDisables?: boolean,
+    ruleSeverities: Object,
+    quiet?: boolean,
+  },
+}*/
+
+/*:: type emptyPostcssResultT = {
```
Comment (jeddy3) [inline lib/lintSource.js]: Dupe?

File: lib/needlessDisables.js [hunk 3]
```diff
@@ -3,21 +3,49 @@
...
+  start: number,
+  used?: boolean,
+}*/
+
+/*:: type unusedRangeT = {
+  start: number,
+  end?: number,
+}*/
+
+/*:: type rangeDataType = {
```
Comment (jeddy3) [inline lib/needlessDisables.js]: Dupe?

File: lib/needlessDisables.js [hunk 4]
```diff
@@ -3,21 +3,49 @@
...
+  all: Array<Object>,
+}
+*/
+
+/*:: type rangeType = {
```
Comment (jeddy3) [inline lib/needlessDisables.js]: Dupe?

File: lib/needlessDisables.js [hunk 5]
```diff
@@ -3,21 +3,49 @@
...
+  end?: number,
+  start: number,
+  used?: boolean,
+}*/
+
+/*:: type unusedRangeT = {
```
Comment (jeddy3) [inline lib/needlessDisables.js]: Dupe?

**Comment by jeddy3** [comment]
> In `lib/needlessDisables.js`, Flow is erroring on line 86 when a "range" is pushed into the `unused.range`. 

Good sleuthing!

It looks like there's [more](https://github.com/stylelint/stylelint/pull/2823#issuecomment-327372045) duplication of code here.

@ivanzusko Can you try updating the PR to:

```
export type stylelint$needlessDisablesReport = Array<{
  source: string,
  ranges: Array<{
    start: number,
    end?: number,
  }>,
}>
```

Am I right that there are seemingly more dupes? I've marked them in review.

**Comment by CAYdenberg** [comment]
There are more errors in the other two files, but I think it might be better to try and get the number down. @ivanzusko if you can eliminate those dupes and make the change suggested above, I'll totally help track down the others ...

**Comment by CAYdenberg** [comment]
Superseded by #2923 - closing.

**Comment by ntwb** [comment]
Thanks once again for the pull request @ivanzusko, sorry this took so long to resolve, it's great to now have @CAYdenberg on board who'll be able to help out with Flow related issues hopefully in a more timely manner than we've been previously handling them 😄

**Comment by ivanzusko** [comment]
@CAYdenberg , @jeddy3 , @ntwb i'm so sorry, guys, for being so slow... had too much stuff besides coding 😞

**Comment by jeddy3** [comment]
No worries :)

**Pull request closed without merging**